src/Controllers/MapController.ts

```diff
@@ -868,24 +882,44 @@ module StreamStats.Controllers {
                     //listeners active during drawing              
                     this.lineDelineationstart = (e) => {
                         if (coordinates.point1.lat === null && coordinates.point1.long === null && coordinates.point2.lat === null && coordinates.point2.long === null) {
+                            console.log('one')
                             coordinates.point1.lat = e.latlng.lat;
                             coordinates.point1.long = e.latlng.lng;
                         } else if (coordinates.point1.lat !== null && coordinates.point1.long !== null && coordinates.point2.lat === null && coordinates.point2.long === null) {
+                            console.log('two')
                             coordinates.point2.lat = e.latlng.lat;
                             coordinates.point2.long = e.latlng.lng;
                             // We have enough points (2) to create a line
-                            const line = [
+                            const lineCoordinates = [
                                 [ coordinates.point1.lat, coordinates.point1.long ],
                                 [ coordinates.point2.lat, coordinates.point2.long ]
                             ];
-                            L.polyline(line, {color: 'blue'}).addTo(map);
-                            //remove listeners
+                            // add line to map 
+                            this.delineationLine = L.polyline(lineCoordinates, {color: 'blue'}).addTo(map)
+
+                            // check line length 
+                            console.log(this.drawControl._getMeasurementString())
+                            var distance = this.drawControl._getMeasurementString();
+                            if (distance.replace(/[^0-9]/g, "") > 13200) { // line is longer than 2.5 miles
+                                this.studyArea.resetDelineationButtons();
+                                map.removeLayer(this.delineationLine)
+                                // remove listeners
+                                map.off("click", this.lineDelineationstart);
+                                this.drawControl.disable();
+                                // throw error
+                                this.toaster.pop("error", "Error", "Delineation not possible. Line is too long, must be shorter than 2.5 miles.", 0);

```

Comment:
Just a suggestion to make the text shorter 

```suggestion
                                this.toaster.pop("error", "Error", "Delineation not possible. Line must be shorter than 2.5 miles.", 0);

```